.pre-commit-config.yaml

```diff
@@ -79,7 +79,8 @@ repos:
             narwhals/_utils\.py|
             narwhals/stable/v./_?dtypes.py|
             narwhals/.*__init__.py|
-            narwhals/.*typing\.py
+            narwhals/.*typing\.py|
+            narwhals/functions.py
```

Comment:
Ok I have two ways to go about this one:

- Whatever is public in `functions.py` will be re-implemented in `stable/v*/__init__.py` where we are already allowed to import the types.
- The alternative is to have the `_impl_<some_function>` in `functions.py` and implement it in `narwhals/__init__.py` for the main namespace so that `Int64` is already imported

Reply:
Mmmh I have mixed feelings about it but I will tag along and see how it looks like 😂

Reply:
I don't really have a preference on this - aside deciding how to pass the default dtype, I think the PR is ready for review (within its currently limited scope)